File: agenttel-node/src/baseline/provider.ts
import type { OperationBaseline } from '../interfaces.js';

/** Returns baseline data for a named operation. */
export interface BaselineProvider {
  getBaseline(operationName: string): OperationBaseline | undefined;
}

/** Returns baselines from a pre-configured map. */
export class StaticBaselineProvider implements BaselineProvider {
  private readonly baselines: Map<string, OperationBaseline>;

  constructor(baselines: Record<string, OperationBaseline>) {
    this.baselines = new Map(Object.entries(baselines));
  }

  getBaseline(operationName: string): OperationBaseline | undefined {
    return this.baselines.get(operationName);
  }
}

File: agenttel-go/baseline/provider.go
// Package baseline provides baseline providers for tracking expected operational norms.
package baseline

import (
	"go.agenttel.dev/agenttel-go/models"
)

// Provider returns baseline data for a named operation.
type Provider interface {
	GetBaseline(operationName string) (models.OperationBaseline, bool)
}

// StaticProvider returns baselines from a pre-configured map.
type StaticProvider struct {
	baselines map[string]models.OperationBaseline
}

// NewStaticProvider creates a StaticProvider from a map of operation baselines.
func NewStaticProvider(baselines map[string]models.OperationBaseline) *StaticProvider {
	m := make(map[string]models.OperationBaseline, len(baselines))
	for k, v := range baselines {
		m[k] = v
	}
	return &StaticProvider{baselines: m}
}

// GetBaseline returns the baseline for the given operation, if configured.
func (s *StaticProvider) GetBaseline(operationName string) (models.OperationBaseline, bool) {
	b, ok := s.baselines[operationName]
	return b, ok
}
